### PR TITLE
Replace travis with CircleCI

### PR DESCRIPTION
**Pull request by marmoure**
This is a minimal config to replace TravisCI with CircleCI.
This PR does not include `coveralls` as it will be added in another PR in the future.
[Coversall](https://www.npmjs.com/package/coveralls) is responsible for showing the code coverage in a nice web UI, we had to remove it as it was relying on TravisCI.
https://github.com/evolvedbinary/jdita/blob/cab9e3a11a18b2d79d1ad1a802b34002cb0c36c3/package.json#L16

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  node-executor:
+    docker:
+      - image: circleci/node:12
+jobs:
+  build_and_test:
+    executor: node-executor
+    steps:
+      - checkout
+      - run: yarn install
+      - run: yarn coverage
```
Comment (adamretter) [inline .circleci/config.yml:12]: Should this be `yarn test` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: 2.1
+executors:
```
Comment (adamretter) [inline .circleci/config.yml:2]: The executors can just be inline for now, we only need move them to their own section if we are going to reuse them for other jobs. As we only have one job we don't need it to be separate. So the following is enough in the job.

```
jobs:
  build-and-test:
  docker:
     - image: cimg/node:16.10
    steps:
      - checkout
      - run: yarn install
      - run: yarn coverage
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  node-executor:
```
Comment (adamretter) [inline .circleci/config.yml:3]: Sometimes you are naming with `-` and sometimes with `_`. Can you choose one and stick with it please? The CircleCI examples seem to use `-`, so perhaps use that one?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    docker:
+      - image: circleci/node:12
```
Comment (adamretter) [inline .circleci/config.yml:5]: We should be specific about which version of Node, e.g. `cimg/node:12.13.1`

**Review by adamretter — CHANGES_REQUESTED** [review]
I think we can simplify this. Maybe also consider using the CircleCI Orb for Node.js to simplify things further, e.g.:

```yaml
version: 2.1

orbs:
  node: circleci/node@5.1.0

jobs:
  build-and-test:
    docker:
      - image: cimg/node:12.13.1
    steps:
      - checkout
      - node/install-packages
          pkg-manager: yarn
      - run:
          name: Run tests
          command: yarn test
```

**Comment by adamretter** [comment]
> [Coversall](https://www.npmjs.com/package/coveralls) is responsible for showing the code coverage in a nice web UI, we had to remove it as it was relying on TravisCI.

As far as I am aware, it is not at all related to Travis CI. What was the problem that you were seeing?

File: .circleci/config.yml (hunk 1 above)
Comment (marmoure) [reply .circleci/config.yml:12]: `yarn test` is included in `yarn coverage`
  https://github.com/evolvedbinary/jdita/blob/cab9e3a11a18b2d79d1ad1a802b34002cb0c36c3/package.json#L15C1-L15C1

**Comment by marmoure** [comment]
> > [Coversall](https://www.npmjs.com/package/coveralls) is responsible for showing the code coverage in a nice web UI, we had to remove it as it was relying on TravisCI.
> 
> As far as I am aware, it is not at all related to Travis CI. What was the problem that you were seeing?

@adamretter when i checked the web ui https://coveralls.io/github/evolvedbinary/jdita
It said running the build via `travis-ci-com` this is what made me say it's dependent on travis-ci
Also when we try to run it locally it throws an error about the missing travis-ci token

**Comment by adamretter** [comment]
@marmoure From the node coveralls documentation it says that it supports CircleCI - https://github.com/nickmerwin/node-coveralls#circleci-orb, is there anything else stopping you from fixing this?

**Comment by marmoure** [comment]
> @marmoure From the node coveralls documentation it says that it supports CircleCI - https://github.com/nickmerwin/node-coveralls#circleci-orb, is there anything else stopping you from fixing this?

No nothing is stopping us from doing that.
We wanted to submit a minimal PR that we are sure that it would work so we have a stable CI to begin with then we can start trying to add `coversall` to it

**Comment by adamretter** [comment]
@marmoure Ah okay, understood. I will merge this, and then please send a follow-up PR with Coveralls reinstated.

**Comment by adamretter** [comment]
@marmoure Just before I merge, I note you haven't added the CI Badge to the `README.md`. Can you do so please?

**Comment by marmoure** [comment]
@adamretter We opted to using the SVG style for the Circle CI badge
Which one do you prefer?

`svg` version

image


`shield` version

image

**Comment by adamretter** [comment]
@marmoure Yes, SVG style is the better one.

**Pull request merged**